inc/BatteryModel.h

```diff
@@ -197,6 +197,8 @@ class BatteryModel final : public PCOE::PrognosticsModel {
                                              const input_type& u,
                                              const output_type& z) const override;
 
+    output_type eventStateEqn(const state_type& x);
```

Comment:
This should have its own type (event_state_type). It is not returning an output vector